cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1213,6 +1214,26 @@ func checkDataAndSavePairing(rootdir string) (bool, error) {
 	return subtle.ConstantTimeCompare(marker1, marker2) == 1, nil
 }
 
+var (
+	// can be mocked in tests
+	partSrcPollIterations = 120
+	partSrcPollWait       = 500 * time.Millisecond
+)
+
+func waitPartSrc(partSrc string) error {
+	// this is used in tests
+	if partSrcPollIterations == 0 {
+		return nil
+	}
+	for i := 0; i < partSrcPollIterations; i++ {
+		if osutil.FileExists(partSrc) {
+			return nil
+		}
+		time.Sleep(partSrcPollWait)
+	}
+	return fmt.Errorf("cannot mount source %v: no such file or directory", partSrc)
```

Comment:
```suggestion
	return fmt.Errorf("cannot mount source %v: no such file or directory after waiting for %s", partSrc, partSrcPollIterations*partSrcPollWait)
```

Reply:
I updated this too, thanks for the suggestion. I had to modify it slightly as it needs type casting.